tests/transforms/test_tape_expand.py

```diff
@@ -328,3 +329,229 @@ class NonDiffPhaseShift(qml.PhaseShift):
 
         assert new_tape is tape
         spy.assert_not_called()
+
+
+# Custom decomposition functions for testing.
+def custom_cnot(wires):
+    return [
+        qml.Hadamard(wires=wires[1]),
+        qml.CZ(wires=[wires[0], wires[1]]),
+        qml.Hadamard(wires=wires[1]),
+    ]
+
+
+def custom_hadamard(wires):
+    return [qml.RZ(np.pi, wires=wires), qml.RY(np.pi / 2, wires=wires)]
+
+
+# Incorrect, for testing purposes only
+def custom_rx(params, wires):
+    return [qml.RY(params, wires=wires), qml.Hadamard(wires=wires)]
+
+
+class TestCreateCustomDecompExpandFn:
+    """Tests for the gradient expand function"""
+
+    def test_no_custom_decomp(self):
+        """Test that sending an empty dictionary results in no decompositions."""
+
+        def circuit():
+            qml.Hadamard(wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        original_dev = qml.device("default.qubit", wires=3)
+        decomp_dev = qml.device("default.qubit", wires=3, custom_decomps={})
+
+        original_qnode = qml.QNode(circuit, original_dev, expansion_strategy="device")
+        decomp_qnode = qml.QNode(circuit, decomp_dev, expansion_strategy="device")
```

Comment:
Do I understand correctly, `expansion_strategy="device"` is the default and a user doesn't need to specify it, but you want to make sure it is used here?

Reply:
Unfortunately, `expansion_strategy="gradient"` is the default, hence why it is specified here 😬 For variational workflows, this makes more sense as a default (since you will see drastically fewer quantum evals per iteration), but this change does make me nervous.

Reply:
Oh so in the changelog example it just works by accident? 

Hm, I think this is fine, but should be explained? Maybe use the device docstring for this?

But even when `expansion_strategy="gradient"` is used the custom decomposition strategy is respected? It's just that gates may not be expanded in the first place, so nothing gets decomposed, right? I think that is ok, this is a decomposition transform, not a "do not use this gate" one...

Reply:
> But even when expansion_strategy="gradient" is used the custom decomposition strategy is respected?

Yep! It just won't be drawn, since the QNode will not (by default) want to decompose that far.

Reply:
so this is more of a drawing annoyance than anything, I suppose

Reply:
> Oh so in the changelog example it just works by accident?

What do you mean?

Reply:
Oh sorry, I only saw now that you specify the strategy there too, but in `draw`...